hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -4017,13 +4019,27 @@ public boolean isReplicationPeerModificationEnabled() {
    */
   public void decommissionRegionServers(final List<ServerName> servers, final boolean offload)
     throws IOException {
+    this.decommissionRegionServers(servers, offload, false);
+  }
+
+  public void decommissionRegionServers(final List<ServerName> servers, final boolean offload,
+    final boolean matchHostNameOnly) throws IOException {
     List<ServerName> serversAdded = new ArrayList<>(servers.size());
     // Place the decommission marker first.
     String parentZnode = getZooKeeper().getZNodePaths().drainingZNode;
     for (ServerName server : servers) {
       try {
         String node = ZNodePaths.joinZNode(parentZnode, server.getServerName());
-        ZKUtil.createAndFailSilent(getZooKeeper(), node);
+        // Encode whether the host should be decommissioned permanently regardless of its
+        // port + startCode combination or not in the znode's data
+        if (matchHostNameOnly) {
+          LOG.info("Marking the host of '{}' as permanently decommissioned in ZooKeeper",
+            server.getServerName());
+        }
+        byte[] data = DrainedZNodeServerData.newBuilder().setMatchHostNameOnly(matchHostNameOnly)
+          .build().toByteArray();
+        // Create a node with binary data
+        ZKUtil.createAndFailSilent(getZooKeeper(), node, data);
```

Comment:
This should not fail silently. If it cannot create the znode that marks the decommissioning , it should fail the RPC. Let the exception throw -- the existing catch clause appears to handle it appropriately.